File: src/KnightFrank.Antares.Dal/Model/Company/Company.cs
namespace KnightFrank.Antares.Dal.Model.Company
{
    using System;
    using System.Collections.Generic;

    using Enum;

    using KnightFrank.Antares.Dal.Model.User;

    public class Company : BaseEntity
    {
        public string Name { get; set; }
        public string WebsiteUrl { get; set; }
        public string ClientCarePageUrl { get; set; }
        public string Description { get; set; }
        public bool? Valid { get; set; }
        public Guid? CompanyCategoryId { get; set; }
        public EnumTypeItem CompanyCategory { get; set; }
        public Guid? CompanyTypeId { get; set; }
        public EnumTypeItem CompanyType { get; set; }
        public virtual ICollection<CompanyContact> CompaniesContacts { get; set; }
        public Guid? ClientCareStatusId { get; set; }
        public EnumTypeItem ClientCareStatus { get; set; }
        public Guid? RelationshipMangerId { get; set; }
        public virtual User RelationshipManger { get; set; }
    }
}

File: src/wwwroot/app/common/models/business/company/company.ts
/// <reference path="../../../../typings/_all.d.ts" />

module Antares.Common.Models.Business {
    export class Company implements Dto.ICompany {
        id: string = null;
        name: string = null;    
        websiteUrl: string = null;
        clientCarePageUrl: string = null;
        clientCareStatusId: string = null;
        contacts: Contact[] = [];
        companiesContacts: Models.Dto.ICompanyContact[] = [];
        description: string = null;
        companyCategoryId: string = null;
        companyTypeId: string = null;
        valid: boolean = false;
        relationshipManager: User = null;

        constructor(company?: Dto.ICompany) {
            if (company) {
                angular.extend(this, company);

                //TODO: check if can be removed
                if (company.contacts && !company.companiesContacts) {
                    this.contacts = company.contacts.map((contact: Dto.IContact) =>{ return new Contact(contact) });
                }

                if (company.companiesContacts && !company.contacts) {
                    this.contacts = company.companiesContacts.map((item) =>{ return new Contact(item.contact) });
                }

                if (company.relationshipManager) {
                    this.relationshipManager = new User(company.relationshipManager);
                }
            }
        }
    }
}